src/lib.rs

```diff
@@ -1,7 +1,87 @@
+//! authenticated (e.g. cryptographically signed) messages. The nodes provide a user-defined type
+//! of _input_ - _transactions_ - to the instance, and handle networking:
+//! Messages received from other nodes must be passed into the instance, and messages produced by
+//! the instance need to be sent to the corresponding nodes.
+//! The algorithm's output are _batches_ of transactions. The order and content of these
+//! batches is guaranteed to be the same in all the nodes, as long as enough nodes are honest and
+//! functioning.
+//!
+//! * It is _Byzantine fault tolerant_: It works correctly even if an adversary controls a number
+//! `f` of the nodes, as long as the total number `N` of nodes is greater than `3 * f`. The
+//! adversary is assumed to be unable to break cryptography, but otherwise can send any kind of
+//! message through their nodes.
+//! * It is also _asynchronous_: It does not make timing assumptions about message delivery. The
+//! adversary can even be allowed to fully control the network scheduling, i.e. arbitrarily delay
+//! messages, even those sent from an honest node to another honest node, as long as every message
+//! eventually arrives. In particular, the algorithm is independent of time scales: it works via
+//! fiber optics and via smoke signals, with the same code and the same parameters.
+//!
+//! Consensus algorithms are a fundamental building block of resilient systems that are distributed
+//! across a number `N > 3 * f` of machines, and that can tolerate any kind of failure (including
+//! complete takeover by an attacker) in up to `f` machines. Use-cases include distributed
+//! databases and blockchains.
+//!
+//! Honey Badger is modular, and composed of several other algorithms that can also be used
+//! independently. All of them are Byzantine fault tolerant and asynchronous.
+//!
+//! ## Algorithms
+//!
+//! [**Honey Badger BFT**](honey_badger/index.html)
+//!
+//! The nodes input any number of _transactions_ (any user-defined type) and outputs a sequence of
+//! _batches_. The batches have sequential numbers (_epochs_) and contain a set of transactions
+//! that were input by the nodes. The sequence and contents of the batches will be the same in all
+//! nodes.
+//!
+//! [**Common Subset**](common_subset/index.html)
+//!
+//! Each node inputs one item. The output is a set of at least `N - f` nodes' items and will be the
+//! same in every node.
+//!
+//! This is the main building block of Honey Badger: In each epoch, every node proposes a number of
+//! transactions. Using the Common Subset protocol, they agree on at least `N - f` of those
+//! proposals. The batch contains the union of these sets of transactions.
+//!
+//! [**Reliable Broadcast**](broadcast/index.html)
+//!
+//! One node, the _proposer_, inputs an item, and every node receives that item as an output. Even
+//! if the proposer is faulty it is guaranteed that either all nodes output the same item or none
+//! at all.
```

Comment:
The disjunction is rather more tautological: either all correct nodes output the item, or no correct nodes output it. I think that "none at all" is not correct here because faulty nodes may do whatever they like!